pilot/pkg/config/kube/crdclient/gen/types.go.tmpl

```diff
@@ -69,6 +70,22 @@ func update(ic versionedclient.Interface, sc serviceapisclient.Interface, cfg co
 	}
 }
 
+func updateStatus(ic versionedclient.Interface, sc serviceapisclient.Interface, cfg config.Config, objMeta metav1.ObjectMeta) (metav1.Object, error) {
+    switch cfg.GroupVersionKind {
+    {{- range . }}
+    	case collections.{{ .VariableName }}.Resource().GroupVersionKind():
+    		return {{.Client}}.{{ .ClientGroupPath }}().{{ .ClientTypePath }}({{if .Namespaced}}cfg.Namespace{{end}}).UpdateStatus(context.TODO(), &{{ .ClientImport }}.{{ .Kind }}{
+    			ObjectMeta: objMeta,
+    			Spec:       *(cfg.Spec.(*{{ .APIImport }}.{{ .Kind }}{{ .TypeSuffix }})),
```

Comment:
Does the `UpdateStatus` method here for client-go modify the spec _and_ status?

Reply:
https://github.com/istio/client-go/blob/master/pkg/clientset/versioned/typed/networking/v1alpha3/workloadentry.gen.go#L139

Looks like this would only affect the subresource.